src/Tasks.UnitTests/GenerateBindingRedirects_Tests.cs

```diff
@@ -259,6 +259,22 @@ public void AppConfigInvalidIfDependentAssemblyNodeIsEmpty()
             redirectResults.Engine.AssertLogContains("MSB3835");
         }
 
+        [Fact]
+        public void AppConfigWhenFilePlacedInLocationWithGB18030Characters()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                TransientTestFolder rootTestFolder = _env.CreateFolder(); 
+                TransientTestFolder testFolder = _env.CreateFolder(Path.Combine(rootTestFolder.Path, "𬴂龨苘爫麤υ㏑䡫3"));
+                string appConfigContents = WriteAppConfigRuntimeSection(string.Empty, testFolder);
+                string outputAppConfigFile = _env.ExpectFile(".config").Path;
```

Comment:
You're creating a new TestEnvironment but using the class-level one. Should this be

```suggestion
                TransientTestFolder rootTestFolder = env.CreateFolder(); 
                TransientTestFolder testFolder = env.CreateFolder(Path.Combine(rootTestFolder.Path, "𬴂龨苘爫麤υ㏑䡫3"));
                string appConfigContents = WriteAppConfigRuntimeSection(string.Empty, testFolder);
                string outputAppConfigFile = env.ExpectFile(".config").Path;
```

?